### PR TITLE
feat(menu): add menu item focus styles

### PR DESCRIPTION
**Pull request by bmdalex**
# Menu

This PR main purposes:
- adds keyboard focus styles by leveraging existing functionality;
- updates existing styles to adhere better to Teams theme;

### TODO

- [x] Conformance test
- [x] Minimal doc site example
- [ ] Stardust base theme
- [x] Teams Light theme
- [ ] Teams Dark theme
- [ ] Teams Contrast theme
- [x] Confirm RTL usage
- [ ] [W3 accessibility](https://www.w3.org/standards/webdesign/accessibility) check
- [ ] [Stardust accessibility](https://github.com/stardust-ui/accessibility) check
- [ ] Update glossary props table
- [x] Update the CHANGELOG.md

# Description

## `type="primary"`

All primary (`type="primary"`) menus get updated background for `active`, `focus` and `hover` states.
screen shot 2018-09-27 at 02 49 14

## `underlined`

Underlined menus get new styles for `active`, `focus` and `hover` states.

screen shot 2018-09-27 at 02 49 46
screen shot 2018-09-27 at 02 50 02

## `iconOnly`

Menus with only icons as item get new styles for `active`, `focus` and `hover` states.

#### 1. regular:
screen shot 2018-09-27 at 02 50 22

#### 2. inverted (by tweeking variables):
screen shot 2018-09-27 at 02 51 37

#### 3. vertical (with `vertical` prop):
screen shot 2018-09-27 at 02 51 57

### CODE REVIEW

**Comment by bmdalex** [comment]
ready for review:
@miroslavstastny @mnajdova @levithomason @kuzhelov

File: src/themes/teams/components/Menu/menuItemStyles.ts [hunk 1]
```diff
@@ -140,94 +166,108 @@ const menuItemStyles: IComponentPartStylesInput<IMenuItemProps, IMenuVariables>
...
+              }),
+
+              ...(underlined && { fontWeight: 700 }),
+
+              ...(underlined && active && underlinedItem(v.defaultActiveColor)),
+            }),
+      }),
+
+      ':focus': {
+        outline: 0,
```
Comment (miroslavstastny) [inline src/themes/teams/components/Menu/menuItemStyles.ts:257]: From now on, any menu other than `iconOnly` and `underlined` has no focus indicator visible :-(

**Review by miroslavstastny — COMMENTED** [review]
Code looks good, not sure about removing focus ring for all Menu types.

File: src/themes/teams/components/Menu/menuItemStyles.ts (hunk 1 above)
Comment (mnajdova) [reply src/themes/teams/components/Menu/menuItemStyles.ts:257]: Agreed with @miroslavstastny we should still use outline: 0 for the variants of the Menu for which we don't provide some other styling.

Comment (alinais) [reply src/themes/teams/components/Menu/menuItemStyles.ts:257]: You should keep the outline for the examples that are not related to the Teams theme, or update it accordingly for the Teams Menu examples.

Comment (bmdalex) [reply src/themes/teams/components/Menu/menuItemStyles.ts:257]: I think there is a little misunderstanding here; All examples **do** have focus indicator visible and that's precisely the reason I took out the outline here at line 257. See DEMOs below for a basic vertical menu:

#### current master:
no focus indicator apart from the blue outline
master

#### this branch:
new focus indicator is the background color => outline is removed
every example has some sort of focus indicator; however, some examples have the same styling for both focused and active states which can be a little misleading
new

**tl;dr**
I don't thing we should do any more changes at this point

Comment (bmdalex) [reply src/themes/teams/components/Menu/menuItemStyles.ts:257]: @miroslavstastny @mnajdova @alinais

Comment (mnajdova) [reply src/themes/teams/components/Menu/menuItemStyles.ts:257]: I think this is enough for now, especially because these menus are not used in the teams theme. We can style them differently if we will receive more details spec about how they should look.

File: src/components/RadioGroup/RadioGroupItem.tsx [hunk 2]
```diff
@@ -32,15 +32,23 @@ export interface IRadioGroupItemProps {
...
 /**
  * @accessibility
  * Radio items need to be grouped in RadioGroup component to correctly handle accessibility.
  */
-class RadioGroupItem extends AutoControlledComponent<Extendable<IRadioGroupItemProps>, any> {
+class RadioGroupItem extends AutoControlledComponent<
+  Extendable<IRadioGroupItemProps>,
+  IRadioGroupItemState
+> {
   static create: Function
```
Comment (mnajdova) [inline src/components/RadioGroup/RadioGroupItem.tsx:52]: Why are these changes part of this PR? Are the Radio used in some Menu's examples?

**Review by mnajdova — APPROVED** [review]
Left one comment for consideration, other then that, looks good by me. Please create an issue for the isFromKeyboard service explaining the problem you encountered.

**Review by alinais — APPROVED** [review]

File: src/components/RadioGroup/RadioGroupItem.tsx (hunk 2 above)
Comment (bmdalex) [reply src/components/RadioGroup/RadioGroupItem.tsx:52]: side effects of `isFromKeyboard` changes

**Review by miroslavstastny — APPROVED** [review]
